gen/feudal

```diff
@@ -36,13 +36,9 @@ our $A_F_FEUDAL_ENTITLEMENT_SUFIX;      *A_F_FEUDAL_ENTITLEMENT_SUFIX =   \'urn:
 #main data structure
 my $userInfo = {};
 
-my %facilityAttributes = attributesToHash $data->getAttributes;
-
-foreach my $resourceData ($data->getChildElements) {
-	my %resourceAttrs = attributesToHash $resourceData->getAttributes;
-
-	foreach my $groupData (($resourceData->getChildElements)[0]->getChildElements) {
-		my $groupMembersLogins = processGroupData $groupData;
+foreach my $resourceId ($data->getResourceIds()) {
+	foreach my $groupId ($data->getGroupIdsForResource(resource => $resourceId)) {
+		processGroupData $groupId, $resourceId;
 	}
 }
 
```

Comment:
Please add `->canonical` to JSON print, so its sorted and comparable between script runs.

Reply:
Added